packages/appkit/src/adapters/ChainAdapterBlueprint.ts

```diff
@@ -0,0 +1,426 @@
+import type { CaipAddress, CaipNetwork, ChainNamespace } from '@reown/appkit-common'
+import type { ChainAdapterConnector } from './ChainAdapterConnector.js'
+import type { Connector as AppKitConnector, Tokens } from '@reown/appkit-core'
+import type UniversalProvider from '@walletconnect/universal-provider'
+import type { W3mFrameProvider } from '@reown/appkit-wallet'
+import { ConstantsUtil, PresetsUtil } from '@reown/appkit-utils'
+import type { AppKitOptions } from '../utils/index.js'
+import type { AppKit } from '../client.js'
+
+type EventName = 'disconnect' | 'accountChanged' | 'switchNetwork'
+type EventData = {
+  disconnect: () => void
+  accountChanged: { address: string; chainId?: number | string }
+  switchNetwork: { address?: string; chainId: number | string }
+}
+type EventCallback<T extends EventName> = (data: EventData[T]) => void
+
+/**
+ * Abstract class representing a chain adapter blueprint.
+ * @template Connector - The type of connector extending ChainAdapterConnector
+ */
+export abstract class AdapterBlueprint<
+  Connector extends ChainAdapterConnector = ChainAdapterConnector
+> {
+  public namespace: ChainNamespace | undefined
+  public caipNetworks?: CaipNetwork[]
+  public projectId?: string
+
+  protected avaiableConnectors: Connector[] = []
+  protected connector?: Connector
+  protected provider?: Connector['provider']

```

Comment:
I think we are using the abstraction wrong here as `Connector['provider']` is `unknown` and we should setup generics for this.